src/jit/lower.cpp

```diff
@@ -462,6 +462,17 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
             hiOp2->gtNext      = hiResult;
             hiResult->gtPrev   = hiOp2;
 
+            if (oper == GT_ADD || oper == GT_SUB)
+            {
+                if (loResult->gtOverflow())
+                {
+                    hiResult->gtFlags |= GTF_OVERFLOW;
```

Comment:
If you don't clear GTF_OVERFLOW from loResult then codegen will very likely generate an overflow check for the lo part at the end of `genCodeForBinary`. That's not right, the overflow check should be generated only for the hi part, for the lo part the overflow condition is meaningless.